### DISCUSSION TITLE
Duplicate logs

### DISCUSSION THREAD

**Discussion by Arbiter-S** [sqlalchemy/sqlalchemy#11011] [Usage Questions] [answered]
### The problem
I'm getting duplicate logs. One goes through my root logger that I configured using logging.basicConfig and one is being passed to another stream handler which goes to stdout.

### How to recreate

This the logging configurations I have.

```
logging.basicConfig(
    format="%(asctime)s - %(name)s - %(levelname)s - %(message)s", level=logging.INFO
)

logging.getLogger("httpx").setLevel(logging.WARNING) # unrelated
logging.getLogger('sqlalchemy.engine').setLevel(logging.INFO)
```
I also have echo set to False.

**Comment by zzzeek** [comment, marked as answer]
hi -

well let's try it

```
import logging

logging.basicConfig( format="%(asctime)s - %(name)s - %(levelname)s - %(message)s", level=logging.INFO )
logging.getLogger("httpx").setLevel(logging.WARNING)
logging.getLogger('sqlalchemy.engine').setLevel(logging.INFO)

from sqlalchemy import create_engine

e = create_engine("sqlite://")

with e.connect() as conn:
    conn.exec_driver_sql("select 'hello world'")
```

output:

```
2024-02-14 09:48:44,472 - sqlalchemy.engine.Engine - INFO - BEGIN (implicit)
2024-02-14 09:48:44,472 - sqlalchemy.engine.Engine - INFO - select 'hello world'
2024-02-14 09:48:44,472 - sqlalchemy.engine.Engine - INFO - [raw sql] ()
2024-02-14 09:48:44,472 - sqlalchemy.engine.Engine - INFO - ROLLBACK

```

no dupes.   so...something else is going on.   maybe an `.echo = True` in there that is being reset - echo=True does a basic config.  so the program above will double log if you do this:

```
e = create_engine("sqlite://", echo=True)
e.echo = False

```

so basically, look around for anything involving echo anywhere and remove it, then also look for duplicate log handlers

**Reply by Arbiter-S** [reply]
Thanks. It seems like I had engines in the files I imported that had echo set to True even though I checked engine.echo and it was correctly set to False, this caused the unexpected behavior.
